visualization/dashboard/src/MLIDashboard/Cohort.ts

```diff
@@ -0,0 +1,128 @@
+import { IFilter, FilterMethods } from "./Interfaces/IFilter";
+import { JointDataset } from "./JointDataset";
+import { ModelExplanationUtils } from "./ModelExplanationUtils";
+
+export class Cohort {
+    public static CohortKey: string = "Cohort";
+    private static _cohortIndex: number = 0;
+
+    public rowCount: number = 0;
+    private readonly cohortIndex: number;
+    private mutateCount: number = 0;
+    private _filteredData: Array<{[key: string]: number}>;
+    private _cachedAverageImportance: number[];
+    private _cachedTransposedLocalFeatureImportances: number[][];
+    private currentSortKey: string | undefined;
+    private currentSortReversed: boolean = false;
+    constructor(public name: string, private jointDataset: JointDataset, public filters: IFilter[] = []) {
+        this.cohortIndex = Cohort._cohortIndex;
+        this.name = name;
+        Cohort._cohortIndex += 1;
+        this.applyFilters();
+    }
+
+    public updateFilter(filter: IFilter, index?: number): void {
+        if (index === undefined) {
+            index = this.filters.length;
+        } 
+
+        this.filters[index] = filter;
+        this.applyFilters();
+    }
+
+    // An id to track if a change requireing rerender has occured.
+    public getCohortID(): number {
+        return this.cohortIndex;
+    }
+
+    public deleteFilter(index: number): void {
+        this.filters.splice(index, 1);
+        this.applyFilters();
+    }
+
+    public getRow(index: number): {[key: string]: number} {
+        return {...this.jointDataset.dataDict[index]}
+    }
+
+    public sort(columnName: string = JointDataset.IndexLabel, reverse?: boolean): void {
+        if (this.currentSortKey !== columnName) {
+            this._filteredData.sort((a, b) => {
+                return a[columnName] - b[columnName];
+            });
+            this.currentSortKey = columnName;
+            this.currentSortReversed = false;
+        }
+        if (this.currentSortReversed !== reverse) {
+            this._filteredData.reverse();
+        }
+    }
+
+    // whether to apply bins is a decision made at the ui control level,
+    // should not mutate the true dataset. Instead, bin props are preserved
+    // and applied when requested.
+    // Bin object stores array of upper bounds for each bin, return the index
+    // if the bin of the value;
```

Comment:
minor: finish comment
     // if the bin of the value;

Reply:
yeah, should be 'of' not if